eppo_core/src/event_ingestion/event_dispatcher.rs

```diff
@@ -0,0 +1,219 @@
+        ));
+
+        (sender, receiver)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::event_ingestion::vec_event_queue::{VecEventQueue, VecEventQueueConfig};
+    use crate::timestamp::now;
+    use serde::Serialize;
+    use serde_json::json;
+    use tokio::time::Duration;
+    use uuid::Uuid;
+    use wiremock::http::Method;
+    use wiremock::matchers::{body_json, method, path};
+    use wiremock::{Mock, MockServer, ResponseTemplate};
+
+    #[derive(Debug, Clone, Serialize)]
+    struct LoginPayload {
+        pub user_id: String,
+        pub session_id: String,
+    }
+
+    #[tokio::test]
+    async fn test_dispatch_starts_delivery() {
+        let event = new_test_event();
+        let mock_server = MockServer::start().await;
+        let mut eppo_events = Vec::new();
+        eppo_events.push(serde_json::to_value(event.clone()).unwrap());
+        let expected_body = json!({"eppo_events": eppo_events });
+        Mock::given(method("POST"))
+            .and(path("/"))
+            .and(body_json(&expected_body))
+            .respond_with(ResponseTemplate::new(200))
+            .mount(&mock_server)
+            .await;
+        run_dispatcher_task(event, mock_server.uri()).await;
+        let received_requests = mock_server.received_requests().await.unwrap();
+        assert_eq!(received_requests.len(), 1);
+        let received_request = &received_requests[0];
+        assert_eq!(received_request.method, Method::POST);
+        assert_eq!(received_request.url.path(), "/");
+        let received_body: serde_json::Value =
+            serde_json::from_slice(&received_request.body).expect("Failed to parse request body");
+        assert_eq!(received_body, expected_body);
+    }
+
+    #[tokio::test]
+    async fn test_dispatch_failed() {
+        let event = new_test_event();
+        let mock_server = MockServer::start().await;
+        let mut eppo_events = Vec::new();
+        eppo_events.push(serde_json::to_value(event.clone()).unwrap());
+        let expected_body = json!({"eppo_events": eppo_events });
+        let response_body =
+            ResponseTemplate::new(200).set_body_json(json!({"failed_events": [event.uuid]}));
+        Mock::given(method("POST"))
+            .and(path("/"))
+            .and(body_json(&expected_body))
+            .respond_with(response_body)
+            .mount(&mock_server)
+            .await;
+        let queue = run_dispatcher_task(event.clone(), mock_server.uri()).await;
+        let received_requests = mock_server.received_requests().await.unwrap();
+        assert_eq!(received_requests.len(), 1);
+        let failed_events = queue.next_batch(QueuedEventStatus::Failed);
+        // assert failed event was moved to failed queue
+        assert_eq!(
+            failed_events,
+            vec![QueuedEvent {
+                event,
+                attempts: 1,
+                status: QueuedEventStatus::Failed
+            }]
+        );
```

Comment:
This assert is failing because we don't use an explicit queue. If we implement retries inline, I think we can just drop the queue and queued events

Reply:
for now I just updated it to assert on the `failure` vec in the receiver https://github.com/Eppo-exp/eppo-multiplatform/pull/159/commits/235e096537940635311d61750284fa10555dbcf9#diff-cc0578a9294bc56ba4e2187fc2ce89c0da05c725b894fe4b8e57ef7d28ffb2c7

Reply:
not sure how to handle retries yet with this architecture, so I'll punt for now